homeassistant/components/media_player/__init__.py

```diff
@@ -548,54 +520,132 @@ def turn_on(self):
+        """Turn the media player on."""
+        yield from self.hass.loop.run_in_executor(
+            None, self.turn_on)
+
     def turn_off(self):
         """Turn the media player off."""
         raise NotImplementedError()
 
+    @asyncio.coroutine
+    def async_turn_off(self):
+        """Turn the media player off."""
+        yield from self.hass.loop.run_in_executor(
+            None, self.turn_off)
+
     def mute_volume(self, mute):
         """Mute the volume."""
         raise NotImplementedError()
 
+    @asyncio.coroutine
+    def async_mute_volume(self, mute):
+        """Mute the volume."""
+        yield from self.hass.loop.run_in_executor(
+            None, self.mute_volume, mute)
+
     def set_volume_level(self, volume):
         """Set volume level, range 0..1."""
         raise NotImplementedError()
 
+    @asyncio.coroutine
+    def async_set_volume_level(self, volume):
+        """Set volume level, range 0..1."""
+        yield from self.hass.loop.run_in_executor(
+            None, self.set_volume_level, volume)
+
     def media_play(self):
         """Send play commmand."""
         raise NotImplementedError()
 
+    @asyncio.coroutine
+    def async_media_play(self):
+        """Send play commmand."""
+        yield from self.hass.loop.run_in_executor(
+            None, self.media_play)
+
     def media_pause(self):
         """Send pause command."""
         raise NotImplementedError()
 
+    @asyncio.coroutine
+    def async_media_pause(self):
+        """Send pause command."""
+        yield from self.hass.loop.run_in_executor(
+            None, self.media_pause)
+
     def media_stop(self):
         """Send stop command."""
         raise NotImplementedError()
 
+    @asyncio.coroutine
+    def async_media_stop(self):
+        """Send stop command."""
+        yield from self.hass.loop.run_in_executor(
+            None, self.media_stop)
+
     def media_previous_track(self):
         """Send previous track command."""
         raise NotImplementedError()
 
+    @asyncio.coroutine
+    def async_media_previous_track(self):
+        """Send previous track command."""
+        yield from self.hass.loop.run_in_executor(
+            None, self.media_previous_track)
+
     def media_next_track(self):
         """Send next track command."""
         raise NotImplementedError()
 
+    @asyncio.coroutine
+    def async_media_next_track(self):
+        """Send next track command."""
+        yield from self.hass.loop.run_in_executor(
+            None, self.media_next_track)
+
     def media_seek(self, position):
         """Send seek command."""
         raise NotImplementedError()
 
-    def play_media(self, media_type, media_id):
+    @asyncio.coroutine
+    def async_media_seek(self, position):
+        """Send seek command."""
+        yield from self.hass.loop.run_in_executor(
+            None, self.media_seek, position)
+
+    def play_media(self, media_type, media_id, **kwargs):
```

Comment:
The problem here is that in the original implementation, all subclasses override this method to include `**kwargs`, and the original implementation assumes it's present here: https://github.com/home-assistant/home-assistant/pull/4951/files#diff-5611ccdb60e42595186058f82df9c7a8L402

Since the service is now calling `async_play_media`, which isn't overridden for most subclasses, which will delegate to the subclass `play_media` method, they both need to be aware of the potential `**kwargs`.